src/Microsoft.ML.Recommend/MatrixFactorizationPredictor.cs

```diff
@@ -0,0 +1,403 @@
+//------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Float = System.Single;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.Recommend;
+using Microsoft.ML.Runtime.Recommend.Internal;
+
+[assembly: LoadableClass(typeof(MatrixFactorizationPredictor), null, typeof(SignatureLoadModel),
+    "Matrix Factorization Predictor Executor", MatrixFactorizationPredictor.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Recommend
+{
+    public sealed class MatrixFactorizationPredictor : IPredictor, ICanSaveModel, ICanSaveInTextFormat, IMIValueMapper, ISchemaBindableMapper,
+        IUserHistoryToItemsRecommender
+    {
+        public const string LoaderSignature = "MFPredictorExec";
```

Comment:
>MFPredictorExec [](start = 47, length = 15)

Minor nit - This class is MFPredictor, I would prefer the MFPredictor for the signature, exec sounds like I can execute something.

Reply:
Sure.

---
In reply to: [225353016](https://github.com/dotnet/machinelearning/pull/1263#discussion_r225353016) [](ancestors = 225353016)